### ISSUE TITLE
Bug: Today Not function API Visualization (Table and DataView) Data from Row not exist

### ISSUE THREAD

**Issue by orwant** [google/google-visualization-issues#1883]
```
What steps will reproduce the problem? Please provide a link to a
demonstration page if at all possible, or attach code.

Since a few months the API has been perfect, but Today, Not function.
I Use DataTable and DataView, The Query function ok, I can access to my information
using getDataTable(), but the Draw method not Draw the values.

My View show me a table with all the rows and the pager, but the rows are blank.

I tryed to use Element inspector give me the next code:
<tr class="google-visualization-table-tr-odd-nonstrict">
  <td class="tableCell google-visualization-table-seq">2</td>
  <td class="tableCell google-visualization-table-td-number">&nbsp;</td>
  <td class="tableCell">&nbsp;</td><td class="tableCell">&nbsp;</td>
  <td class="tableCell">&nbsp;</td><td class="tableCell">&nbsp;</td>
  <td class="tableCell google-visualization-table-td-center">&nbsp;</td>
  <td class="tableCell">&nbsp;</td><td class="tableCell">&nbsp;</td>
</tr>

But I use a click listener and get the informacion from the DataTable from the Row
and All the information is here...

¿Why I can not see the information in the DataView?


var spreadsheetKey = "MySpreadSheetKey";

google.load('visualization', '1.0', {'packages':['table']});
google.setOnLoadCallback(inicializar);

var url = 'http://spreadsheets.google.com/tq?key='+spreadsheetKey+'&sheet=Citas';
var queryCadena = queryBase+" ";
var query, options, container, cssClassNames;


function inicializar(){
  container = document.getElementById('resultados');
  query = new google.visualization.Query(url+"&tq="+encodeURI(queryCadena));
  query.send(cargarVisualizacion);
}


function cargarVisualizacion(response){
  if (response.isError()) {
    errorDialog('Error in query: ' + response.getMessage() + ' ' + response.getDetailedMessage());
    return;
  }
  var data = response.getDataTable();
  var table = new google.visualization.Table(container);
  var view = new google.visualization.DataView(data);
  view.setColumns([
    {label: 'Date', sourceColumn: 0, type: 'number'},
    {label: 'Location', sourceColumn: 1, type: 'string'},
    {label: 'Guest', sourceColumn: 4, type: 'string'},
    {label: 'Place', sourceColumn: 6, type: 'string'},
    {label: 'Phone', sourceColumn: 7, type: 'string'},
    {label: 'Date', sourceColumn: 9, type: 'date'},
    {label: 'Time One', sourceColumn: 10, type: 'string'},
    {label: 'Time tow', sourceColumn: 11, type: 'string'}
  ]);
  table.draw(view, {
    page: 'enable',
    pageSize: 5,
    showRowNumber: true,
    allowHTML: true,
    cssClassNames: {
      tableCell : 'tableCell'
    }
  });

  google.visualization.events.addListener(table, 'select', function() {
    var row = table.getSelection()[0].row;
    listarInvitados(data.getValue(row,12), data.getValue(row,0));
  });
  $("#enviarDatos").prop("disabled", true);
}


/*

CSS STYLE -------------
.tableCell{
  padding: 0px;
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 5px;
  padding-bottom: 5px;
}

*/

function listarInvitados(param1, param2){
  alert(param1);
  alert(param2);
}



What component is this issue related to (PieChart, LineChart, DataTable,
Query, etc)?

DataTable and DataView


Are you using the test environment (version 1.1)?
(If you are not sure, answer NO)
Version 1.0
google.load('visualization', '1.0', {'packages':['table']});


What operating system and browser are you using?
Windows: Chrome 40.0..., FireFox 34
Linux: Chrome 39, Firefox 32


*********************************************************
For developers viewing this issue: please click the 'star' icon to be
notified of future changes, and to let us know how many of you are
interested in seeing it resolved.
*********************************************************

```

Original issue reported on code.google.com by `oraygoza@infoanalitica.com` on 2015-02-20 00:52:04

**Comment by orwant** [comment]
```
Check the DOM and see whether the content actually exists. We had the same problem and
noticed that some tables where getting their heights changed to 1px and width to 0px.
But not all tables and not all always both properties.

overriding the height and width with css !important fixed it up easily enough for our
purposes.
```

Original issue reported on code.google.com by `brigge0001` on 2015-02-20 00:57:10

**Comment by orwant** [comment]
```
Thanks, but it is not the same bug, as we can see the DataView fill each row with &nbsp,
but the DataTable showme the information, I check it with and a click listener and
show the Row data in alert window.

<tr class="google-visualization-table-tr-odd-nonstrict">
  <td class="tableCell google-visualization-table-seq">2</td>
  <td class="tableCell google-visualization-table-td-number"> </td>
  <td class="tableCell"> </td><td class="tableCell"> </td>
  <td class="tableCell"> </td><td class="tableCell"> </td>
  <td class="tableCell google-visualization-table-td-center"> </td>
  <td class="tableCell"> </td><td class="tableCell"> </td>
</tr>

Tnx
```

Original issue reported on code.google.com by `oraygoza@infoanalitica.com` on 2015-02-20 01:05:18

**Comment by orwant** [comment]
```
Ah, so it isn't. Sorry - didn't read your initial comment properly.
```

Original issue reported on code.google.com by `brigge0001` on 2015-02-20 01:11:10

**Comment by orwant** [comment]
```
The problem oraygoza reported turns out to be due to a couple problems, both involving
AppsScript.  The single window sandbox mode of AppsScript is converting the Date values
into strings, as output by date.toString().  You might be able to use the special Google
Charts string date notation instead, which looks like "Date(yyyy, mm, dd)" (with 0-based
month).   Another AppsScript-related issue is that when you use a DataView, this triggers
a bug in Google Charts in which we were modifying the columns array, but AppsScript
makes this be an immutable array, so it fails.  We will fix this latter problem today,
but the Date problem remains.

Alternatively, you may be able to use the iframe sandbox mode in AppsScript to avoid
all the problems caused by the single window mode.
```

Original issue reported on code.google.com by `dlaliberte@google.com` on 2015-02-26 14:47:55

**Comment by orwant** [comment]
```
Hi Daniel, thank you very much.

To solve the problem it the date format we use a "calc" option to
parse.ToString the cell value, into the addColum parameters. ¿What do you
thnik about it? What kind of problems could we have?

About the problem with apps script, We change the Sanboxmode from NATIVE to
IFRAME and its fix the problem.

thank you very much.

2015-02-26 8:48 GMT-06:00 <google-visualization-api-issues@googlecode.com>:

-- 

 Omar Raygoza
Coordinador CIDET Acapulco
 T: (55) 1205-6900
 E: oraygoza@infoanalitica.com



Este mensaje y los datos adjuntos son para uso exclusivo de la persona
o entidad a la que expresamente ha sido enviada y puede contener
informacion PRIVILEGIADA y CONFIDENCIAL.
```

Original issue reported on code.google.com by `oraygoza@infoanalitica.com` on 2015-02-27 14:59:41

**Comment by orwant** [comment]
```
When using the AppsScript native sandbox mode, your workaround for the Date object,
to specify a calc propertly in the DataView to parse the date string, is fine.  Using
the date string notation should also work.

The problem I mentioned before, in which using a DataView results in all fields appearing
to have no value, should be fixed now (in 1.0).  But the Date problem still exists
because it is AppsScript that is incorrectly converting dates to strings, when using
native sandbox mode.

If you change to use iframe sandbox mode, then you won't have to worry about converting
the Date yourself, and everything should work as is.



```

Original issue reported on code.google.com by `dlaliberte@google.com` on 2015-02-27 15:48:30
